test/test_torch.py

```diff
@@ -229,27 +229,33 @@ def test_linear_algebra_scalar_raises(self):
         self.assertRaises(RuntimeError, lambda: torch.addr(m, v, s))
         self.assertRaises(RuntimeError, lambda: torch.addr(m, s, v))
 
-    def _testMath(self, torchfn, mathfn, large=True):
-        def _testMathSize(size, self, torchfn, mathfn):
+    def _testMath(self, torchfn, mathfn, large=True, precs=(1e-8, 1e-4)):
+        def _testMathSize(size, self, torchfn, mathfn, dtype, prec):
             # contiguous
-            m1 = torch.randn(*size)
+            m1 = torch.randn(*size).type(dtype)
             res1 = torchfn(m1[4])
             res2 = res1.clone().zero_()
             for i, v in enumerate(m1[4]):
                 res2[i] = mathfn(v.item())
-            self.assertEqual(res1, res2)
+            self.assertEqual(res1, res2, prec)
 
             # non-contiguous
-            m1 = torch.randn(*size)
+            m1 = torch.randn(*size).type(dtype)
             res1 = torchfn(m1[:, 4])
             res2 = res1.clone().zero_()
             for i, v in enumerate(m1[:, 4]):
                 res2[i] = mathfn(v.item())
-            self.assertEqual(res1, res2)
-        _testMathSize((10, 5), self, torchfn, mathfn)
-        if large:
-            # Trigger parallelism
-            _testMathSize((10, 50000), self, torchfn, mathfn)
+            self.assertEqual(res1, res2, prec)
+
+        types = [
+            ('torch.DoubleTensor', precs[0]),
+            ('torch.FloatTensor', precs[1]),
+        ]
+        for (dtype, prec) in types:
```

Comment:
Loops make it hard to debug what failed. You're only testing floats and double so just unroll it:

```python
_testMathSize((10, 5), self, torchfn, mathfn, torch.double, precs[0])
_testMathSize((10, 5), self, torchfn, mathfn, torch.float, precs[1])
if large:
 _testMathSize((10, 50000), self, torchfn, mathfn, torch.double, precs[0])
 _testMathSize((10, 50000), self, torchfn, mathfn, torch.float, precs[1])
```